tfsdklog/options.go

```diff
@@ -1,101 +1,59 @@
 package tfsdklog
 
 import (
-	"io"
 	"os"
 	"strings"
 
 	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/terraform-plugin-log/internal/logging"
 )
 
-// Option defines a modification to the configuration for a logger.
-type Option func(loggerOpts) loggerOpts
-
-type loggerOpts struct {
-	name            string
-	level           hclog.Level
-	includeLocation bool
-
-	// some private options to be used only by tflog for testing purposes
-	// we should never export an Option that sets these
-	output      io.Writer
-	includeTime bool
-}
-
-func applyLoggerOpts(opts ...Option) loggerOpts {
-	// set some defaults
-	l := loggerOpts{
-		name:            "sdk",
-		includeLocation: true,
-		includeTime:     true,
-		output:          os.Stderr,
-	}
-	for _, opt := range opts {
-		l = opt(l)
-	}
-	return l
-}
-
-func withOutput(output io.Writer) Option {
-	return func(l loggerOpts) loggerOpts {
-		l.output = output
-		return l
-	}
-}
-
-func withoutTimestamp() Option {
-	return func(l loggerOpts) loggerOpts {
-		l.includeTime = false
-		return l
-	}
-}
-
 // WithLogName returns an option that will set the logger name explicitly to
 // `name`. This has no effect when used with NewSubsystem.
-func WithLogName(name string) Option {
-	return func(l loggerOpts) loggerOpts {
-		l.name = name
+func WithLogName(name string) logging.Option {
+	return func(l logging.LoggerOpts) logging.LoggerOpts {
+		l.Name = name
 		return l
 	}
 }
 
+// WithStderrFromInit returns an option that tells the logger to write to the
```

Comment:
Nit: Not sure if we would want to keep this at the end of the file to reduce VCS "churn" and keep it alphabetical.